### ISSUE TITLE
How to detect bad XML file

### ISSUE THREAD

**Issue by tomaszlabackikainos** [jahewson/node-big-xml#8]
Hi, after researching possibilities both in the node-big-xml and in the node-expat I still don't figure out how could I detect that file given to the reader is actually a bad file (e.g. image instead of proper xml)?

**Comment by jahewson** [comment]
It turns out the node-expat has an unusual API and rather than throw errors or emit an 'error' event, its parser.parse function returns a boolean indicating success. Further error details can then be fetched from parser.getError().

To fix this, I've added error handling in aca148107ec8df16b9ea86030dbc8845f5bd06d1. BigXmlReader now emits an 'error' event when the underlying gzip or expat parsers encounter an error. Here's what happens now if you try to parse a .png file:

```                                                                                                                                    
Error: XML Error: not well-formed (invalid token)                                                                                                 
    at ReadStream.<anonymous> (/node-big-xml/lib/big-xml.js:28:26)                                                       
    at emitOne (events.js:96:13)                                                                                                                  
    at ReadStream.emit (events.js:188:7)                                                                                                          
    at readableAddChunk (_stream_readable.js:176:18)                                                                                              
    at ReadStream.Readable.push (_stream_readable.js:134:10)                                                                                      
    at onread (fs.js:1984:12)                                                                                                                     
    at FSReqWrap.wrapper [as oncomplete] (fs.js:681:17)      
```

Thanks for reporting this!

**Comment by jahewson** [comment]
Published to npm in `v1.0.0`.

**Comment by tomaszlabackikainos** [comment]
Great work John, awesome response time, thanks!
